docs/mode-reference.rst

```diff
@@ -89,6 +91,50 @@ disableAutodetect
 Disables autodetection for this language.
 
 
+compilerExtensions (USE WITH CAUTION)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- **type**: an array of compiler extensions ie: ``(mode, parentMode) -> {} ``
+
+This allows grammars to extend the mode compiler to add their own syntactic
+sugar to make reading and writing grammars easier.  **Note: This is heavily
+dependent upon compiler internals and may NOT be stable from minor release to
+minor release.** *It is currently recommended only for 1st party grammars.* The
+intention is that we use grammars to "test" out new compiler extensions and if
+they perform well promote them into the core library.
+
+mode
+  The incoming mode object
+
+parent
```

Comment:
Perhaps `parentMode` is better, to match above `(mode, parentMode) -> {}`.

Reply:
Done.